polyfills/settings/settings.js

```diff
@@ -136,55 +118,72 @@
   var _observers = {};
 
   // And this is something else that might be reusable...
-  function Observer(setting, callback) {
+  function Observer(reqId, extraData) {
     this._id = null;
+    _observers[extraData.settingName][extraData.callback] = this;
     this.serialize = () => {
       if (!this._id) {
-        this._id = ++_currentRequestId;
+        this._id = reqId;
       }
       return {
         id: this._id,
-        data: {
-          operation: 'addObserver',
```

Comment:
again, addObserver is what this object is. Just do a 
data: {operation: 'addObserver', settingName: extraData.setting}